sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/credential/AccessTokenCache.java

```diff
@@ -23,28 +23,27 @@
 /**
  * A token cache that supports caching a token and refreshing it.
  */
-public class AccessTokenCacheImpl {
+public class AccessTokenCache<T extends TokenCredential, R extends TokenRequestContext> {
```

Comment:
This class is an implementation detail and ideally should not be exposed to users publicly.

Reply:
Thanks Vinay. I fully understand. This was just a draft to point out the following

1. Most policies will like to cache tokens as tokens are generally valid for a good chunk of time. Each policy should right fully have their caches as internal but the type can be public.
2. Most token credentials would need to fetch tokens based on a context as the token is likely to vary per request.
3. Having a cache that lets you work on a generic token credential/toeknrequestcontext makes it possible for it to be reused. We should discuss the right way to do it but I see value in exposing this functionality in our core library.

Reply:
Thanks for the update.
To reuse the cache, I feel it can be done via the policy internally itself, Generics are probably not required to reuse it. Possibly I am missing some context, so we can discuss offline on a quick call.